consensus/service/src/consensus_service.rs

```diff
@@ -636,70 +646,165 @@ impl<
-                    latest_block_hash = None;
-                    latest_block_timestamp = None;
-                    blocks_behind = None;
-                }
+                    }?;
+
+                (highest_block, highest_block_signature)
+            };
+
+            let (is_behind, highest_peer_block_index): (bool, u64) = {
+                let byzantine_ledger_arc_once_cell: Arc<OnceCell<ByzantineLedger>> =
+                    match byzantine_ledger.upgrade() {
+                        Some(instance) => Ok(instance),
+                        None => {
+                            log::error!(logger, "ByzantineLedger has been dropped.");
+                            Err(RpcStatus::new(
+                                RpcStatusCode::UNAVAILABLE,
+                                Some("ByzantineLedger has been dropped".to_string()),
+                            ))
+                        }
+                    }?;
+
+                let byzantine_ledger_instance = match byzantine_ledger_arc_once_cell.get() {
+                    Some(instance) => Ok(instance),
+                    None => {
+                        log::error!(logger, "ByzantineLedger has not been initialized.");
+                        Err(RpcStatus::new(
+                            RpcStatusCode::UNAVAILABLE,
+                            Some("ByzantineLedger has not been initialized".to_string()),
+                        ))
+                    }
+                }?;
+
+                let is_behind: bool = byzantine_ledger_instance.is_behind();
+                let highest_peer_block_index: u64 = byzantine_ledger_instance.highest_peer_block();
+                (is_behind, highest_peer_block_index)
             };
-            Ok(json!({
+
+            let network_config = config.network();
+
+            let json: String = config_and_status_as_json(
+                &config,
+                &network_config,
+                &highest_block,
+                highest_block_signature.as_ref(),
+                is_behind,
+                highest_peer_block_index,
+            );
+
+            Ok(json)
+        })
+    }
+}
+
+impl<
+        E: ConsensusEnclave + Clone + Send + Sync + 'static,
+        R: RaClient + Send + Sync + 'static,
+        TXM: TxManager + Clone + Send + Sync + 'static,
+    > Drop for ConsensusService<E, R, TXM>
+{
+    fn drop(&mut self) {
+        let _ = self.stop();
+    }
+}
+
+/// Format the service's configuration and ledger status as JSON.
+///
+/// # Arguments
+/// * `config` - ConsensusService configurations.
+/// * `network_config` - Network configuration.
+/// * `highest_block` - Highest block in the local ledger.
+/// * `highest_block_signature` - Signature, if any, for the highest block in the local ledger.
+/// * `is_behind` - True if the local ledger is behind the peers' ledgers.
+/// * `highest_peer_block_index` -  Highest block index agreed upon by peers.
+fn config_and_status_as_json(
```

Comment:
This is a pure function that renders the configuration and status data as JSON. It is currently difficult to unit test ConsensusService, so extracting some of the logic like this makes it easier to test.